core/src/main/java/nl/nn/adapterframework/http/rest/ApiServiceDispatcher.java

```diff
@@ -205,11 +207,33 @@ protected JsonObject generateOpenApiJsonSchema(Collection<ApiDispatchConfig> cli
 			}
 		}
 		root.add("paths", paths.build());
-		root.add("components", Json.createObjectBuilder().add("schemas", schemas));
+		JsonObjectBuilder components = Json.createObjectBuilder();
+		components.add("schemas", schemas);
+		root.add("components", components);
 
 		return root.build();
 	}
 
+	private JsonArrayBuilder mapServers(HttpServletRequest request) {
+		JsonArrayBuilder serversArray = Json.createArrayBuilder();
+		String protocol = request.isSecure() ? "https://" : "http://";
+		String suffix = "/api";
```

Comment:
Could you retrieve the suffix from this property `servlet.ApiListenerServlet.urlMapping` and fall back to the default (api) when not set?